src/Functions/registerFunctions.cpp

```diff
@@ -1,145 +1,60 @@
-void registerFunctionsConversion(FunctionFactory &);
-void registerFunctionCastOrDefault(FunctionFactory &);
-void registerFunctionsDateTime(FunctionFactory &);
-void registerFunctionsEmbeddedDictionaries(FunctionFactory &);
-void registerFunctionsExternalDictionaries(FunctionFactory &);
-void registerFunctionsExternalModels(FunctionFactory &);
-void registerFunctionsFormatting(FunctionFactory &);
-void registerFunctionHashID(FunctionFactory &);
-void registerFunctionsHashing(FunctionFactory &);
-void registerFunctionsHigherOrder(FunctionFactory &);
-void registerFunctionsLogical(FunctionFactory &);
-void registerFunctionsMiscellaneous(FunctionFactory &);
-void registerFunctionsRandom(FunctionFactory &);
-void registerFunctionsReinterpret(FunctionFactory &);
-void registerFunctionsRound(FunctionFactory &);
-void registerFunctionsString(FunctionFactory &);
-void registerFunctionsStringArray(FunctionFactory &);
-void registerFunctionsStringSearch(FunctionFactory &);
-void registerFunctionsStringRegexp(FunctionFactory &);
-void registerFunctionsStringSimilarity(FunctionFactory &);
-void registerFunctionsStringTokenExtractor(FunctionFactory &);
-void registerFunctionsURL(FunctionFactory &);
-void registerFunctionsVisitParam(FunctionFactory &);
-void registerFunctionsMath(FunctionFactory &);
-void registerFunctionsGeo(FunctionFactory &);
-void registerFunctionsIntrospection(FunctionFactory &);
-void registerFunctionsNull(FunctionFactory &);
-void registerFunctionsJSON(FunctionFactory &);
-void registerFunctionsSQLJSON(FunctionFactory &);
-void registerFunctionToJSONString(FunctionFactory &);
-void registerFunctionsConsistentHashing(FunctionFactory & factory);
-void registerFunctionsUnixTimestamp64(FunctionFactory & factory);
-void registerFunctionBitHammingDistance(FunctionFactory & factory);
-void registerFunctionTupleHammingDistance(FunctionFactory & factory);
-void registerFunctionsStringHash(FunctionFactory & factory);
-void registerFunctionValidateNestedArraySizes(FunctionFactory & factory);
-void registerFunctionsSnowflake(FunctionFactory & factory);
-void registerFunctionTid(FunctionFactory & factory);
-void registerFunctionLogTrace(FunctionFactory & factory);
-void registerFunctionsTimeWindow(FunctionFactory &);
-void registerFunctionToBool(FunctionFactory &);
-void registerFunctionMinSampleSize(FunctionFactory &);
-
-// meilisearch
-void registerFunctionMeiliMatch(FunctionFactory & factory);
-
-#if USE_SSL
-void registerFunctionEncrypt(FunctionFactory & factory);
-void registerFunctionDecrypt(FunctionFactory & factory);
-void registerFunctionAESEncryptMysql(FunctionFactory & factory);
-void registerFunctionAESDecryptMysql(FunctionFactory & factory);
-void registerFunctionShowCertificate(FunctionFactory &);
-#endif
+
+void earlyRegisterFunctionReplicate(FunctionFactory & factory);
+void earlyRegisterFunctionsLogical(FunctionFactory & factory);
 
 void registerFunctions()
 {
+    auto symbols = DB::SymbolIndex::instance()->symbols();
```

Comment:
Maybe we can put the detection logic into `Symbolndex`, without the additional startup cost.

Reply:
I am not quite understand. We still need to build the symbols upfront right?

Reply:
I mean you can try adding another field like `function_symbols` and collect it along with https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/SymbolIndex.cpp#L249